kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -228,6 +228,7 @@ andEqualsFunctions = fmap mapEqualsFunctions
     , (BothT,   \_ _ _ -> domainValueAndConstructorErrors, "domainValueAndConstructorErrors")
     , (BothT,   \_ _ _ -> domainValueAndEqualsAssumesDifferent, "domainValueAndEqualsAssumesDifferent")
     , (BothT,   \_ _ _ -> stringLiteralAndEqualsAssumesDifferent, "stringLiteralAndEqualsAssumesDifferent")
+    , (AndT,    \_ _ _ -> distributeAnd, "distributeAnd")
```

Comment:
The new unification function needs to take a `TermSimplifier` argument like `Builtin.Map.unifyEquals`.